capstone/src/main/java/com/google/sps/servlets/JoinServlet.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import com.google.sps.data.Lesson;
+import com.google.sps.data.Room;
+import com.google.sps.data.User;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.sps.data.Lesson;
+import com.google.sps.data.Room;
+import com.google.sps.data.RequestParser;
+import com.google.sps.service.DatabaseService;
+import com.google.sps.data.RequestParser;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/join")
+public class JoinServlet extends HttpServlet {
+
+    private static final String JOIN_PROPERTY_KEY = "join";
+    private static final String UNJOIN_PROPERTY_KEY = "unjoin";
+    private static final String ROOM_ID_PROPERTY_KEY = "room_id";
+    private static final String ACTION_PROPERTY_KEY = "action";
```

Comment:
These are not property keys. The last one is a query parameter, the room_id one is a property key, and the first two are just action values. Since the join action can only have one of a set number of possible values, use an enum